drivers/hmis/spec/jobs/merge_clients_job_spec.rb

```diff
@@ -0,0 +1,182 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+
+RSpec.describe Hmis::MergeClientsJob, type: :model do
+  # Probably other specs aren't cleaning up:
+  before(:all) { Hmis::Hud::Client.with_deleted.destroy_all }
```

Comment:
We do this in most of the tests:


```
  before(:all) do
    cleanup_test_environment
  end
  after(:all) do
    cleanup_test_environment
  end

```

Reply:
Thanks. I'll check it out.

Reply:
I would like to know why I end up with clients that weren't created in the merge clients spec even though I'm destroying all clients here. And only on CI. It's very strange. Maybe rails is swallowing the "do them all" and I need to truncate? I'm hoping not to be so heavy handed, as doing this all over the place surely makes the suite much longer. These calls might account for 30 minutes of the suite time by my calculations. I guess if we need it, we need it, though.

Reply:
Yeah, I don't know. I think there are a bunch of tests that don't clean up resources, but they were really hard to track down, so instead we just added this heavy handed cleanup in every single spec? Its kind of before my time, and I don't know what the appropriate way to handle test data is in rpsec.
Are you still seeing extra clients present despite running the cleanup?